controller/src/action/action_repository.rs

```diff
@@ -1,17 +1,54 @@
+use serde::{Deserialize, Serialize};
 use sqlx::PgPool;
 use std::sync::Arc;
 
-use crate::parser::pipe_parser::Type;
+use crate::{
+    action::action_service::ActionDTO, grpc_scheduler::ActionStatus, parser::pipe_parser::Type,
+};
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct Action {
     pub id: i64,
     pub pipeline_id: i64,
     pub name: String,
-    container_uri: String,
+    pub container_uri: String,
```

Comment:
What is container uri ?

Reply:
It corresponds to the container registry address pointing to the image (e.g. `debian/bullseye-slim:latest`)